modules/serverless-resolving-operator-upgrade-failure.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+//
+// * /serverless/admin_guide/serverless-upgrades.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-resolving-operator-upgrade-failure_{context}"]
+= Resolving {ServerlessOperatorName} upgrade failure
+
+In some cases, such as when you do manual uninstalls and reinstalls, you might encounter an error when upgrading {ServerlessOperatorName}. In this situation, you need to manually reinstall {ServerlessOperatorName}.
+
+.Procedure
+
+. Identify the version of {ServerlessOperatorName} that was installed originally using the `project.yaml` files for the different releases.
++
+For example, the error message during attempted upgrade contains this string:
++
+[source,yaml]
+----
+The installed KnativeServing version is v0.25.1.
+----
++
+In this case, the KnativeServing version is 0.25.1, which corresponds to {ServerlessOperatorName} 1.19.
+
+. Uninstall {ServerlessOperatorName} and all its install plans.
+
+. Manually install {ServerlessOperatorName} of version discovered in the first step.
```

Comment:
In this step we need to provide an example Subscription otherwise it will be hard to figure this out (one has to specify startingCSV). The example below should be fine:

Create serverless-subscription.yaml with the following content:
```
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: serverless-operator
  namespace: openshift-serverless
spec:
  channel: stable
  name: serverless-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
  installPlanApproval: Manual
  startingCSV: serverless-operator.v1.26.0
```
Install the subscription:
```
oc apply -f serverless-subscription.yaml
```